crates/sui-bridge/src/sui_mock_client.rs

```diff
@@ -51,7 +51,7 @@ impl SuiMockClient {
     pub fn default() -> Self {
         Self {
             chain_identifier: "".to_string(),
-            latest_checkpoint_sequence_number: 0,
+            latest_checkpoint_sequence_number: Arc::new(Mutex::new(0)),
```

Comment:
not super important, but I think an `Arc<AtomicU64>` here would allow you to avoid the mutex